### PR TITLE
As an app user, I want a verse preview for cross refs, so I can more quickly compare related verses

### PR DESCRIPTION
**Pull request by danbenn**
### Summary

Need to get verse plaintext for verse search results inside the app.

Also, when expanding the cross references section, need to show verse previews for each cross reference.

### How to Test

Tested end-to-end.

Also has unit tests

### CODE REVIEW

File: server/src/Bible.controller.ts [hunk 1]
```diff
@@ -57,6 +57,22 @@ export class BibleController {
...
+    ) {
+        const verseContent = await this.bibleEngine.getPlaintextForReferenceRange({
+            versionUid,
+            bookOsisId: osisId,
+            versionChapterNum: chapterNr,
+            versionVerseNum: verseNr
+        });
+        return { verseContent };
+    }
+
```
Comment (chriswep) [inline server/src/Bible.controller.ts]: maybe add

```
    @Post('/ref/plaintext')
    getReferenceRange(@Body() ref: IBibleReferenceRangeQuery) {
        return this.bibleEngine.getPlaintextForReferenceRange(ref);
    }
```

that way you can use it easily for e.g. crossrefs, which none of the other routes support (chapter spanning refs for example) and you don't need to add a plaintext version for every route

File: client/src/Bible.api.ts [hunk 2]
```diff
@@ -1,177 +1,160 @@
+import {
```
Comment (chriswep) [inline client/src/Bible.api.ts]: i wonder why this got reformatted? did you edit it manually? (it should not be edited)

File: server/src/Bible.controller.ts (hunk 1 above)
Comment (chriswep) [reply server/src/Bible.controller.ts]: one could also argue for using an optional query parameter `?plaintext=true`. this way we have one less method (and you would just set the `plaintext` parameter when calling the client method.

File: client/src/BibleEngineClient.ts [hunk 3]
```diff
@@ -64,6 +64,21 @@ export class BibleEngineClient {
...
+    async getPlaintextForReferenceRange(range: any, forceRemote = false) {
+        if (this.localBibleEngine && !forceRemote) {
+            try {
+                return this.localBibleEngine.getPlaintextForReferenceRange(range);
+            } catch (e) {
+                if (!(e instanceof BibleVersionRemoteOnlyError)) throw e;
+            }
+        }
+        if (this.remoteApi) {
+            const response = await this.remoteApi.getVersePlaintext(range);
```
[the remark this answers is not in the document]
Comment (chriswep) [reply client/src/BibleEngineClient.ts]: see the suggested range method in the controller

**Comment by danbenn** [comment]
I'll be writing a new express server for the mobile app, so I'll revisit this code then 👍

**Pull request closed without merging**